<?xml version="1.0" ?>
<launch>
  
  <node pkg="capsicum_detection" type="realsense_registration" name="realsense_registration" output="screen"> </node>


  <node pkg="ros_kinfu" type="ros_kinfu" name="ros_kinfu" output="screen">

    <!-- INPUT Subscriber Topics -->
    <!--param name="depth_image_topic" value="/camera/depth/image_raw" -->
    <param name="depth_image_topic" value="/camera/depth/image_registered" />
    <param name="rgb_image_topic" value="/camera/rgb/image_registered" />
    <param name="camera_info_topic" value="/camera/depth/camera_info" />

     <!-- Output Publisher Topics -->
    <param name="tsdf_topic" value="/ros_kinfu/tsdf/points" />
    <param name="cloud_topic" value="/ros_kinfu/depth/points" />
    <param name="cloud_colour_topic" value="/ros_kinfu/depth_registered/points" />
    <param name="camera_pose_topic" value="/ros_kinfu/camera/pose" />
    <param name="depth_image_topic" value="/ros_kinfu/depth/image" />
    
    <!-- Frames ID -->
    <param name="kf_world_frame_id" value="kf_world" />
    <param name="kf_camera_frame_id" value="kf_camera" />

    <param name="publish_points" value="true" />
    <param name="visualise" value="false" />
    <param name="color_integration" value="true" />
    <param name="use_hints" value="false" />
    <param name="update_kinect_world_frame" value="false" />

    <param name="camera_fx" value="463.888885" />
    <param name="camera_fy" value="463.888885" />
    <param name="camera_cx" value="320.0" />
    <param name="camera_cy" value="240.0" />
    <param name="image_height" value="480" />
    <param name="image_width" value="640" />

    <param name="volume_size" value="0.5" />
    <param name="tsdf_trunc" value="0.005" />
    <param name="icp_weight" value="0.01" />
    <param name="camera_movement_threshold" value="0.001" />
    <param name="gpu_device" value="0" />

  </node>


</launch>
